backend/src/main/java/ca/bc/gov/app/service/client/ClientService.java

```diff
@@ -141,6 +142,79 @@ public Mono<ClientDetailsDto> getClientDetails(
             "Unable to process request. This sole proprietor is not owned by a person"
         )));
   }
+  
+  public Mono<ForestClientDetailsDto> getClientDetailsByClientNumber(
+      String clientNumber,
+      List<String> groups
+  ) {
+      log.info("Loading details for {} for user role {}", clientNumber, groups.toString());
+
+      return legacyService
+          .searchByClientNumber(clientNumber, groups)
+          .flatMap(forestClientDetailsDto -> {
+              String corpRegnNmbr = forestClientDetailsDto.corpRegnNmbr();
+
+              if (corpRegnNmbr == null || corpRegnNmbr.isEmpty()) {
+                  log.info("Corporation registration number not provided. Returning legacy details.");
+                  return Mono.just(forestClientDetailsDto);
+              }
+
+              log.info("Retrieved corporation registration number: {}", corpRegnNmbr);
+
+              return bcRegistryService
+                      .requestDocumentData(corpRegnNmbr)
+                      .next()
+                      .flatMap(documentMono -> 
+                            populateGoodStandingInd(forestClientDetailsDto, 
+                                                    documentMono)
+                      );
+          });
+  }
+
+  private Mono<ForestClientDetailsDto> populateGoodStandingInd(
+      ForestClientDetailsDto forestClientDetailsDto,
+      BcRegistryDocumentDto document
+  ) {
+      Boolean goodStandingInd = document.business().goodStanding();
+      String goodStanding;
+      if (goodStandingInd == null) {
```

Comment:
There's a `BooleanUtils` with a method that generates a yes no string, and even one that can generate strings based on the boolean value. It is a good approach to reduce code, so take this as a suggestion only

Reply:
Done! Thanks for the suggestion